python/lsst/ctrl/oods/guiderList.py

```diff
@@ -0,0 +1,86 @@
+# This file is part of ctrl_oods
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+import logging
+import time
+from typing import List
+
+from lsst.ctrl.oods.guiderEntry import GuiderEntry
+
+LOGGER = logging.getLogger(__name__)
+
+
+class GuiderList(List[GuiderEntry]):
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def remove_by_name(self, guider_file: str):
+        """Remove the matching guider_file name
+
+        Parameters
+        ----------
+        guider_file: `str`
+            name of the guider file to remove from the list
+        """
+        for i, entry in enumerate(self):
+            if entry.guider_resource_path == guider_file:
+                del self[i]
+                return
+
+    def purge_old_entries(self, max_age_seconds: float) -> int:
+        """Remove entries older than the specified number of seconds
+
+        Parameters
+        ----------
+        max_age_seconds: `int`
+            maximum number of seconds the file should be
+
+        Returns
+        -------
+        removed: `List[GuiderEntry]`
+            entries removed from the list
+        """
+        current_time = time.time()
+        cutoff_time = current_time - max_age_seconds
+        kept = GuiderList()
+        removed = GuiderList()
+
+        for entry in self:
+            if entry.timestamp >= cutoff_time:
+                kept.append(entry)
+            else:
+                removed.append(entry)
+
+        # replace in place
+        self[:] = kept
```

Comment:
If you modify the list in place anyway, you may consider using `list.pop(index)`. If I got it right, you could replace lines 64 and 67-74 with something like
```python
for index, entry in enumerate(self):
    if entry.timestamp < cutoff_time:
        removed.append(self.pop(index))
```